neps/nep-0371.md

```diff
@@ -0,0 +1,149 @@
+---
+NEP: 371
+Title: Runtime Function Call Access Key Info
+Author: Ben Kurrek <ben.kurrek@near.org>, Jakob Meier <jakob@near.org>
+DiscussionsTo: https://gov.near.org/t/expose-gas-price-and-contract-access-key-info-within-runtime/24788
+Status: Draft
+Type: Standards Track
+Category: Review
+Created: 13-Jul-2022
+---
+
+## Summary
+
+There is currently no way to query for information about access keys stored on a contract at runtime. Adding the host function `access_key_info` to the runtime specification will allow to query this for the current account an easy way. This fully covers the most important use cases that are currently known for querying access keys at runtime.
+
+## Motivation
+
+As contracts start to use access keys in more creative ways, developers will want to know information such as the access key's allowance, receiver and more. Since contracts are often built for scalability, Gas and storage costs are often delegated to the end user. If you charge the user for an access key and its allowance, when the key is deleted or used up, that allowance should be refunded to the purchaser. Currently, there is no way to query for the leftover allowance before a key is deleted. This information can tell you how much Gas was spent, and allows for a slew of different scenarios to arise.
```

Comment:
> Since contracts are often built for scalability, Gas and storage costs are often delegated to the end user.

I am not sure I understand what scalability has to do with gas and storage costs being delegated to the end user.

Reply:
If you're building for scalability, you generally want your users to front these costs as to not drain your contract.